recipes/gtk/all/conanfile.py

```diff
@@ -0,0 +1,128 @@
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class LibnameConan(ConanFile):
+    name = "gtk"
+    description = "libraries used for creating graphical user interfaces for applications."
+    topics = ("conan", "gtk", "widgets")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gtk.org"
+    license = "LGPL-2.1-or-later"
+    generators = "pkg_config"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_wayland": [True, False],
+        "with_x11": [True, False],
+        "with_pango": [True, False]
+        }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_wayland": False,
+        "with_x11": True,
+        "with_pango": True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os != "Linux":
+            del self.options.with_wayland
+            del self.options.with_x11
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Linux":
+            if self.options.with_wayland or self.options.with_x11:
+                if not self.options.with_pango:
+                    raise ConanInvalidConfiguration("with_pango option is mandatory when with_wayland or with_x11 is used")
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("GTK recipe is not yet compatible with Windows. Contributions are welcome.")
+    
+    def build_requirements(self):
+        self.build_requires("meson/0.54.2")
+        if not tools.which("pkg-config"):
+            self.build_requires("pkgconf/1.7.3")
+    
+    def requirements(self):
+        self.requires("gdk-pixbuf/2.42.0")
+        self.requires("glib/2.67.0")
+        if self.settings.compiler != "Visual Studio":
+            self.requires("cairo/1.17.2")
+        if self.settings.os == "Linux":
+            self.requires("at-spi2-atk/2.38.0")
+            if self.options.with_wayland:
+                self.requires("xkbcommon/0.10.0")
+                self.requires("wayland") # FIXME: Create an actual Wayland package(s)
```

Comment:
TODO: Figuring out which parts of Wayland gtk needs and what packages we need accordingly

Reply:
the last missing one is [wayland-protocols](https://github.com/GNOME/gtk/blob/3.24.22/meson.build#L552) which is blocked by https://github.com/conan-io/conan-center-index/pull/3663
wayland-client, wayland-cursor and wayland-egl are already packaged by https://conan.io/center/wayland